mlx/coverity_services.py

```diff
@@ -1,614 +1,415 @@
+                    "totalRows": 2720,
+                    "columns": [list of column keys]
+                    "rows": list of [list of dictionaries {"key": <key>, "value": <value>}]
+                }
         """
-        logging.info('Querying Coverity for defects in project [%s] stream [%s] ...', project, stream)
-
-        # define the project
-        project_id = self.client.factory.create('projectIdDataObj')
-        project_id.name = project
-
-        # and the stream
-        stream_id = self.client.factory.create('streamIdDataObj')
-        stream_id.name = stream
-
-        # create filter spec
-        filter_spec = self.client.factory.create('snapshotScopeDefectFilterSpecDataObj')
-
-        # only for this stream
-        filter_spec.streamIncludeNameList.append(stream_id)
-
+        logging.info("Querying Coverity for defects in stream [%s] ...", stream)
+        query_filters = [
+            {
+                "columnKey": "streams",
+                "matchMode": "oneOrMoreMatch",
+                "matchers": [
+                    {
+                        "class": "Stream",
+                        "name": stream,
+                        "type": "nameMatcher"
+                    }
+                ]
+            }
+        ]
         # apply any filter on checker names
-        if filters['checker']:
-            self.config_service.get_checkers()
-            self.handle_attribute_filter(filters['checker'],
-                                         'Checker',
-                                         self.config_service.checkers,
-                                         filter_spec.checkerList,
-                                         allow_regex=True)
+        if filters["checker"]:
+            # this should be a keyMatcher (columnKey: checker)
+            filter_values = self.handle_attribute_filter(filters["checker"], "Checker", self.checkers, allow_regex=True)
+            if filter_values:
+                query_filters.append(self.assemble_query_filter("checker", filter_values, "keyMatcher"))
 
         # apply any filter on impact status
-        if filters['impact']:
-            self.handle_attribute_filter(filters['impact'], 'Impact', IMPACT_LIST, filter_spec.impactNameList)
+        if filters["impact"]:
+            # this should be a keyMatcher (columnKey: displayImpact)
+            filter_values = self.handle_attribute_filter(filters["impact"], "Impact", IMPACT_LIST)
+            if filter_values:
+                query_filters.append(self.assemble_query_filter("displayImpact", filter_values, "keyMatcher"))
 
         # apply any filter on issue kind
-        if filters['kind']:
-            self.handle_attribute_filter(filters['kind'], 'Kind', KIND_LIST, filter_spec.issueKindList)
+        if filters["kind"]:
+            # this should be a keyMatcher (columnKey: displayIssueKind)
+            filter_values = self.handle_attribute_filter(filters["kind"], "displayIssueKind", KIND_LIST)
```

Comment:
Shouldn't displayIssueKind be Kind?

Reply:
There is no key `Kind`, only `displayIssueKind`

Reply:
I was wrong, here it's about the name; resolved in edff3124515fecf965401f40c8555b6e829d3ef7